sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -416,7 +415,7 @@ export class EventHubConsumerClient {
         validateEventPositions(options.startPosition);
       }
       ({ targetedPartitionId, eventProcessor } = this.createEventProcessorForSinglePartition(
-        handlersOrPartitionId1,
+        String(handlersOrPartitionId1),
```

Comment:
So are we literally saying we take _anything_ or do we have some constraints? For instance, is it just numbers and strings? Can we encode that into the typescript signature (number | string), for instance?

Reply:
I ask this because whenever I see an arbitrary string-ization I'm curious why.

Reply:
We have no constraints. This follows the model of input validation where we try our best to make the connection/request to the service and service is responsible for giving any errors

Reply:
Can we put a comment on there? I "reverted" it not knowing what the purpose of it was - might be nice for future programmers to know the same.

Reply:
(Consider this entire convo a 'NIT', btw. Feel free to proceed)

Reply:
Done with https://github.com/Azure/azure-sdk-for-js/pull/9521/commits/a0cd3358a19e50b201a5c078bae23c60970f879e